### PR TITLE
Added endpoints for report settings template

### PR DESCRIPTION
**Pull request by s-aLman**
Added below endpoints and test coverage
- api.projects.reports.settings-templates.getMany
- api.projects.reports.settings-templates.get
- api.projects.reports.settings-templates.patch
- api.projects.settings-templates.delete

### CODE REVIEW

**Comment by s-aLman** [comment]
@andrii-bodnar Pushed changes for report settings template endpoints. Did not add the below endpoint, as I was facing some issue with that. Please check and merge if it looks good to you for other endpoints 🙂. Thanks.

- [https://developer.crowdin.com/enterprise/api/v2/#operation/api.projects.reports.settings-templates.post](url)

**Comment by andrii-bodnar** [comment]
@s-aLman thanks a lot for your contribution! 🎉 

What kind of issues you've faced with the `api.projects.reports.settings-templates.post` API endpoint?

**Comment by s-aLman** [comment]
@andrii-bodnar I faced issue with the test case that I wrote for the post endpoint. It was not getting passed. It was related with "No match for request payload ". I checked test case multiple times but no luck.

**Comment by andrii-bodnar** [comment]
@s-aLman please push the code for this endpoint and the test as well, I would like to see this test and error related

**Comment by s-aLman** [comment]
sure, will do that.

**Comment by s-aLman** [comment]
@andrii-bodnar pushed the changes for post endpoint.

File: src/test/java/com/crowdin/client/reports/ReportsApiTest.java [hunk 1]
```diff
@@ -71,4 +80,61 @@ public void downloadReportTest() {
...
+        ReportSettingsTemplateRequest.ProofreadIndividualRate individualRate = new ReportSettingsTemplateRequest.ProofreadIndividualRate();
+        individualRate.setLanguageIds(Collections.singletonList("uk"));
+        individualRate.setUserIds(Collections.singletonList(1));
+        individualRate.setRates(Collections.singletonList(regularRate));
+
+        config.setRegularRates(Collections.singletonList(regularRate));
+        config.setIndividualRates(Collections.singletonList(individualRate));
+
+        ResponseObject<ReportSettingsTemplate> responseObject = this.getReportsApi().addReportSettingsTemplate(projectId, request);
+        assertEquals(responseObject.getData().getId(), projectId);
```
Comment (andrii-bodnar) [inline src/test/java/com/crowdin/client/reports/ReportsApiTest.java:114]: There is no id in the response?

**Comment by andrii-bodnar** [comment]
@s-aLman the API doc is a bit wrong for the "Add Report Settings Template" method and should be fixed. Sorry for the confusion.

This is the real response example for this endpoint:

```json
{
    "data": {
        "id": 1,
        "name": "my report template",
        "currency": "USD",
        "unit": "strings",
        "mode": "simple",
        "config": {
            "regularRates": [
                {
                    "mode": "no_match",
                    "value": 0.1
                }
            ],
            "individualRates": [
                {
                    "languageIds": [
                        "uk"
                    ],
                    "userIds": [
                        20
                    ],
                    "rates": [
                        {
                            "mode": "tm_match",
                            "value": 0.1
                        }
                    ]
                }
            ]
        },
        "createdAt": "2022-10-15T11:57:57+00:00",
        "updatedAt": null
    }
}
```

**Comment by s-aLman** [comment]
@andrii-bodnar I updated the response but still not able to resolve it. I am not sure why it is keep getting failed.

**Comment by andrii-bodnar** [comment]
@s-aLman please push this update

**Comment by s-aLman** [comment]
@andrii-bodnar done. please check.

**Comment by andrii-bodnar** [comment]
@s-aLman checked - please pay attention to the response structure. You can revert your changes, it's not affecting anything.

The main thing - there is the `data` object with the template info and this object has an `id` field also

**Comment by andrii-bodnar** [comment]
You need to change only the `src/test/resources/api/reports/addReportSettingsTemplate.json` file

**Comment by s-aLman** [comment]
@andrii-bodnar yeah, I got that but the response already has data object and the id also in it. Are you saying about the request body because the request does not have the id ?

**Comment by s-aLman** [comment]
@andrii-bodnar okay. Its confirmed now. I will change the request then. Let's hope for the best. Thanks.

**Comment by andrii-bodnar** [comment]
@s-aLman sorry, I meant the response should the same as in my example

File: src/test/java/com/crowdin/client/reports/ReportsApiTest.java [hunk 2]
```diff
@@ -30,8 +34,13 @@ public List<RequestMock> getMocks() {
...
-        );
+                RequestMock.build(this.url + "/projects/" + projectId + "/reports/" + id + "/download", HttpGet.METHOD_NAME, "api/reports/downloadLink.json"),
+                RequestMock.build(this.url + "/projects/" + projectId + "/reports/settings-templates", HttpGet.METHOD_NAME, "api/reports/listReportSettingsTemplate.json"),
+                RequestMock.build(this.url + "/projects/" + projectId + "/reports/settings-templates/" + reportSettingsTemplateId, HttpGet.METHOD_NAME, "api/reports/reportSettingsTemplate.json"),
+                RequestMock.build(this.url + "/projects/" + projectId + "/reports/settings-templates/" + reportSettingsTemplateId, HttpPatch.METHOD_NAME, "api/reports/editReportSettingsTemplate.json", "api/reports/reportSettingsTemplate.json"),
+                RequestMock.build(this.url + "/projects/" + projectId + "/settings-templates/" + reportSettingsTemplateId, HttpDelete.METHOD_NAME),
+                RequestMock.build(this.url + "/projects/" + projectId + "/reports/settings-templates", HttpPost.METHOD_NAME, "api/reports/addReportSettingsTemplate.json", "api/reports/reportSettingsTemplate.json")
```
Comment (andrii-bodnar) [inline src/test/java/com/crowdin/client/reports/ReportsApiTest.java:42]: Probably here you need a separate JSON files for request and response mocking

**Comment by andrii-bodnar** [comment]
@s-aLman first, try to investigate which of two asserts is failing in the `addReportSettingsTemplateTest`

**Comment by s-aLman** [comment]
@andrii-bodnar yes, doing that.

File: src/main/java/com/crowdin/client/reports/ReportsApi.java [hunk 3]
```diff
@@ -51,4 +49,48 @@ public ResponseObject<DownloadLink> downloadReport(Long projectId, String report
...
         return ResponseObject.of(downloadLinkResponseObject.getData());
     }
+
+    /**
+     * @param projectId project identifier
+     * @param limit     maximum number of items to retrieve (default 25)
+     * @param offset    starting offset in the collection (default 0)
+     * @return list of report settings template
+     */
+    public ResponseList<ReportSettingsTemplate> listReportSettingsTemplate (Long projectId, Integer limit, Integer offset) throws HttpException, HttpBadRequestException {
```
Comment (andrii-bodnar) [inline src/main/java/com/crowdin/client/reports/ReportsApi.java:59]:
```suggestion
    public ResponseList<ReportSettingsTemplate> listReportSettingsTemplate(Long projectId, Integer limit, Integer offset) throws HttpException, HttpBadRequestException {
```

Comment (andrii-bodnar) [reply src/main/java/com/crowdin/client/reports/ReportsApi.java:59]: This space is redundant here. Also, please check the rest of added methods

Comment (s-aLman) [reply src/main/java/com/crowdin/client/reports/ReportsApi.java:59]: sure.

Comment (s-aLman) [reply src/main/java/com/crowdin/client/reports/ReportsApi.java]: How about now?

**Comment by andrii-bodnar** [comment]
@s-aLman LGTM, thanks a lot for your contribution! 🚀

**Review by andrii-bodnar — APPROVED** [review]

**Pull request merged**